web/packages/build/package.json

```diff
@@ -21,7 +21,8 @@
     "@storybook/addon-toolbars": "^6.5.16",
     "@storybook/builder-webpack5": "^6.5.16",
     "@storybook/manager-webpack5": "^6.5.16",
-    "@swc/plugin-styled-components": "^1.5.122",
+    "@swc/core": "^1.7.0",
+    "@swc/plugin-styled-components": "^2.0.10",
```

Comment:
I was looking through [the setup instructions for this plugin](https://styled-components.com/docs/tooling#swc-plugin) and I found a comment about swc not supporting `pure`:

https://github.com/gravitational/teleport/blob/fb60c97fb67c7db9c15f23eb7db4a52a64376031/web/packages/build/vite/react.mjs#L36-L53

Then I followed [the link to the Next.js docs](https://nextjs.org/docs/architecture/nextjs-compiler#styled-components) from the same file, which points to https://github.com/vercel/next.js/issues/30802 as to why `pure` is not supported yet. But it seems like this was actually implemented a couple of months ago: https://github.com/swc-project/plugins/pull/238

@ryanclark, do you think we could enable this in prod? I'm not sure if it makes any difference, but hey at least it should be supported now. I didn't want to block the PR on this because enabling that option and upgrading swc are separate issues.